SU2_CFD/include/variables/CVariable.hpp

```diff
@@ -47,2298 +47,2327 @@
 
 #include "../../../Common/include/config_structure.hpp"
 #include "../fluid_model.hpp"
+#include "../../../Common/include/toolboxes/C2DContainer.hpp"
 
 
 using namespace std;
 
+/*--- Forward declarations of some derived CVariable classes ---*/
+/*---
+ Derived CSolvers keep a private pointer of the safest type they can downcast to.
+ This bypasses most of the virtual indirection (within that solver).
+ These declarations avoid having to include all CVariable hpps in
+ solver_structure.hpp, they will not be need when that file is split.
+---*/
+class CEulerVariable;
+class CIncEulerVariable;
+class CTurbVariable;
+class CDiscAdjVariable;
+class CDiscAdjFEABoundVariable;
+
 /*!
  * \class CVariable
  * \brief Main class for defining the variables.
  * \author F. Palacios
  */
 class CVariable {
 protected:
+  using Idx_t = unsigned long;
+  using VectorType = C2DContainer<Idx_t, su2double, StorageType::ColumnMajor, 64, DynamicSize, 1>;
```

Comment:
As requested I spelled out these names, I think they are uncontroversial now.
I'll do a replace all on Idx_t and the end, until then this helps me find new methods added to CVariable in other PR's.